src/Scientist/Internals/Experiment.cs

```diff
@@ -10,7 +10,25 @@ internal class Experiment<T, TClean> : IExperiment<T, TClean>, IExperimentAsync<
 
         private static readonly Func<Task<bool>> _alwaysRun = () => Task.FromResult(true);
         private static readonly Action<Operation, Exception> _alwaysThrow
-            = (operation, exception) => { throw exception; };
+            = (operation, exception) =>
+            {
+                string exceptionMessage;
+                switch (operation)
```

Comment:
Is there ever a case where we _won't_ know the operation in question? Seems to me like this switch isn't needed, and we can always throw with the `exceptionMessage = $"An exception occurred within the experiment during the '{operation.ToString()}' operation.";` message.